File: src/commands/RemoveAllByPostalAddressCommand.java
package commands;

import client.Application;
import commands.core.ServerAndClientSideCommand;
import lib.ExecutionStatus;
import lib.functions.CallbackFunction;
import organization.Address;
import organization.OrganizationManager;
import organization.UserInteractiveOrganizationBuilder;

public class RemoveAllByPostalAddressCommand extends ServerAndClientSideCommand {
    public RemoveAllByPostalAddressCommand(CallbackFunction callbackFunction, Application application, OrganizationManager organizationManager) {
        super(callbackFunction, application, organizationManager);
    }

    @Override
    protected void executeThrowableCommand(String[] args, CallbackFunction callback) throws Exception {
        var organizationBuilder = new UserInteractiveOrganizationBuilder(application.getBufferedReaderWithQueueOfStreams(), false);

        Address address = organizationBuilder.getAddress();
        organizationManager.removeAllByPostalAddress(address);

        callback.invoke(ExecutionStatus.SUCCESS, null);
    }
}

File: src/commands/client_side/RemoveAllByPostalAddressCommand.kt
package commands.client_side;

import application.Application;
import commands.client_side.core.ServerAndClientSideCommand;
import lib.ExecutionStatus;
import commands.client_side.core.ClientCallbackFunction;
import OrganizationDatabase.Address;
import OrganizationDatabase.OrganizationManagerInterface;
import application.UserInteractiveOrganizationBuilder;

public class RemoveAllByPostalAddressCommand extends ServerAndClientSideCommand {
    public RemoveAllByPostalAddressCommand(ClientCallbackFunction clientCallbackFunction, Application application, OrganizationManagerInterface organizationDatabase) {
        super(clientCallbackFunction, application, organizationDatabase);
    }

    @Override
    protected void executeImplementation(String[] args, ClientCallbackFunction callback) throws Exception {
        var organizationBuilder = new UserInteractiveOrganizationBuilder(application.getBufferedReaderWithQueueOfStreams(), false);

        Address address = organizationBuilder.getAddress();
        organizationDatabase.removeAllByPostalAddress(address);

        callback.invoke(ExecutionStatus.SUCCESS, null);
    }
}
